### ISSUE TITLE
Download & Upload-URLs return 401 when reordering query parameters

### ISSUE THREAD

**Issue by olihey** [OneDrive/onedrive-api-docs#792]
The REST client we are using is reordering query parameters for given URLs. This causes trouble for the download and upload URLs provided by Microsoft Graph.

Example:
```
https://raygina-my.sharepoint.com/personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&Translate=false&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InNhc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0&ApiVersion=2.0
```
is the download URL I get from the API and downloading works fine, but my client does some magic and changes the order of the query parameters as visible that the `ApiVersion=2.0` is moved from the end of the URL to the front of the parameters:
```
https://raygina-my.sharepoint.com/personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?ApiVersion=2.0&Translate=false&UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InNhc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0
```
But even though those two URLs should express the same the second one fails with a 401 Unauthorized with the response `{"error":"invalid_request","error_description":"This access token is not valid on this endpoint."}`

Detailed curl logs:
```
olihey@oliwin:~$ curl -v "https://raygina-my.sharepoint.com/personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&Translate=false&tempauth=eyJ0eXAiOiJKV1Q
iLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDA
wMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2l
kIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InN
hc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM
0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0&ApiVersion=
2.0"
*   Trying 13.107.6.151...
* Connected to raygina-my.sharepoint.com (13.107.6.151) port 443 (#0)
* found 148 certificates in /etc/ssl/certs/ca-certificates.crt
* found 592 certificates in /etc/ssl/certs
* ALPN, offering http/1.1
* SSL connection using TLS1.2 / ECDHE_RSA_AES_128_GCM_SHA256
*        server certificate verification OK
*        server certificate status verification SKIPPED
*        common name: *.sharepoint.com (matched)
*        server certificate expiration date OK
*        server certificate activation date OK
*        certificate public key: RSA
*        certificate version: #3
*        subject: C=US,ST=WA,L=Redmond,O=Microsoft Corporation,OU=Microsoft Corporation,CN=*.sharepoint.com
*        start date: Tue, 06 Dec 2016 22:28:44 GMT
*        expire date: Fri, 06 Apr 2018 22:28:44 GMT
*        issuer: C=US,ST=Washington,L=Redmond,O=Microsoft Corporation,OU=Microsoft IT,CN=Microsoft IT SSL SHA2
*        compression: NULL
* ALPN, server accepted to use http/1.1
> GET /personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&Translate=false&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InNhc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0&ApiVersion=2.0 HTTP/1.1
> Host: raygina-my.sharepoint.com
> User-Agent: curl/7.47.0
> Accept: */*
>
< HTTP/1.1 200 OK
< Cache-Control: private
< Content-Length: 2838
< Content-Type: text/plain
< Accept-Ranges: bytes
< ETag: "{30773B25-1106-435F-9B2D-19673DB13AE9},5"
< P3P: CP="ALL IND DSP COR ADM CONo CUR CUSo IVAo IVDo PSA PSD TAI TELo OUR SAMo CNT COM INT NAV ONL PHY PRE PUR UNI"
< X-SharePointHealthScore: 0
< X-Download-Options: noopen
< Content-Disposition: attachment;filename*=utf-8''small%5Ffile%2Etxt;filename="small_file.txt"
< CTag: {30773B25-1106-435F-9B2D-19673DB13AE9},5,2
< X-AspNet-Version: 4.0.30319
< SPRequestGuid: b16d479e-6064-4000-cd12-05507924b648
< request-id: b16d479e-6064-4000-cd12-05507924b648
< MS-CV: nkdtsWRgAEDNEgVQeSS2SA.0
< Strict-Transport-Security: max-age=31536000
< X-FRAME-OPTIONS: SAMEORIGIN
< X-Powered-By: ASP.NET
< MicrosoftSharePointTeamServices: 16.0.0.7324
< X-Content-Type-Options: nosniff
< X-MS-InvokeApp: 1; RequireReadOnly
< X-MSEdge-Ref: Ref A: 14B933C870204EE18BC89706D2444ECC Ref B: AMS04EDGE0515 Ref C: 2018-02-05T06:36:36Z
< Date: Mon, 05 Feb 2018 06:36:36 GMT
<
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Mauris purus dui, aliquet non risus ac, ultricies commodo elit. Integer luctus, dolor volutpat efficitur commodo, diam ipsum consequat leo, eget iaculis velit tortor semper odio. Ut sagittis porta enim eget ullamcorper. Curabitur porttitor magna lorem, eget maximus massa gravida vel. Sed sed magna et justo dignissim venenatis. Nulla dapibus sit amet elit nec sodales. Cras et est non diam ultricies imperdiet. Donec eget dignissim dolor. Duis malesuada eget risus at semper. Quisque cursus tortor ut lectus commodo, id vehicula ipsum tempor.
Aliquam scelerisque arcu at sem egestas sollicitudin. Mauris commodo eros lectus, in blandit lectus accumsan eget. Nam mollis eros magna, eget tempor dolor porta at. Cras sit amet dolor dignissim sapien blandit tempus. In purus lorem, semper eget pulvinar id, efficitur vel nunc. In hac habitasse platea dictumst. Pellentesque sollicitudin metus quis euismod sagittis. Cras aliquam non sem nec suscipit. Aliquam consequat viverra fringilla. Morbi scelerisque quam id laoreet porttitor. Sed vestibulum est id ligula posuere, nec tempus velit congue. Maecenas a enim blandit, condimentum urna eu, sodales nisl. Praesent ut eros fringilla, feugiat urna ut, lacinia sem.
Nunc scelerisque sem in lorem egestas fermentum. Aenean a velit eget magna malesuada fringilla. In sed magna id nunc tempor pharetra in sit amet lectus. Fusce tempor ex sit amet erat pulvinar, nec fringilla sapien ultrices. Nullam risus eros, vehicula vel dolor sit amet, elementum interdum quam. Nunc dictum nec velit ut pretium. Nunc placerat magna ornare dolor ullamcorper aliquet. Aenean sagittis leo vitae sapien auctor, ut iaculis dolor lacinia.
Maecenas imperdiet odio urna, id volutpat nulla gravida ac. Sed volutpat velit ac purus consequat, at laoreet mi blandit. Nam consequat, eros at iaculis blandit, velit dolor porttitor arcu, ac tempus sapien risus id diam. Integer in tincidunt diam. Quisque libero libero, molestie vitae tempor at, luctus quis diam. Maecenas molestie enim eu pellentesque fringilla. Nullam nec massa urna. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis dapibus commodo tortor eget rutrum. Sed vulputate id orci ac bibendum. Fusce nec lectus dui. Aliquam eu sapien diam. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.
Pellentesque vel lacus magna. Donec in lectus in est maximus rutrum. Phasellus aliquet orci libero, ac volutpat ex dapibus vitae. Suspendisse sed* Connection #0 to host raygina-my.sharepoint.com left intact
 ornare ex. Pellentesque ultrices egestas interdum. Suspendisse consectetur pellentesque mauris, a accumsan urna laoreet a. Maecenas quis sodales diam. Fusce pharetra interdum convallis. Vivamus non diam erat. Nullam massa ante, hendrerit at metus vitae, porta consequat ligula.


olihey@oliwin:~$ curl -v "https://raygina-my.sharepoint.com/personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?ApiVersion=2.0&Translate=false&UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InNhc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0"
*   Trying 13.107.6.151...
* Connected to raygina-my.sharepoint.com (13.107.6.151) port 443 (#0)
* found 148 certificates in /etc/ssl/certs/ca-certificates.crt
* found 592 certificates in /etc/ssl/certs
* ALPN, offering http/1.1
* SSL connection using TLS1.2 / ECDHE_RSA_AES_128_GCM_SHA256
*        server certificate verification OK
*        server certificate status verification SKIPPED
*        common name: *.sharepoint.com (matched)
*        server certificate expiration date OK
*        server certificate activation date OK
*        certificate public key: RSA
*        certificate version: #3
*        subject: C=US,ST=WA,L=Redmond,O=Microsoft Corporation,OU=Microsoft Corporation,CN=*.sharepoint.com
*        start date: Tue, 06 Dec 2016 22:28:44 GMT
*        expire date: Fri, 06 Apr 2018 22:28:44 GMT
*        issuer: C=US,ST=Washington,L=Redmond,O=Microsoft Corporation,OU=Microsoft IT,CN=Microsoft IT SSL SHA2
*        compression: NULL
* ALPN, server accepted to use http/1.1
> GET /personal/olihey_raygina_onmicrosoft_com/_layouts/15/download.aspx?ApiVersion=2.0&Translate=false&UniqueId=30773b25-1106-435f-9b2d-19673db13ae9&tempauth=eyJ0eXAiOiJKV1QiLCJhbGciOiJub25lIn0.eyJhdWQiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAvcmF5Z2luYS1teS5zaGFyZXBvaW50LmNvbUBhNTliNzQ3My1lZDJmLTRkOWItODFhZC03ZTYxOGNlNGMzZmEiLCJpc3MiOiIwMDAwMDAwMy0wMDAwLTBmZjEtY2UwMC0wMDAwMDAwMDAwMDAiLCJuYmYiOiIxNTE3ODEyNTA5IiwiZXhwIjoiMTUxNzgxNjEwOSIsImVuZHBvaW50dXJsIjoiTmQ3bVdvcDBVS3d0Z1I0ZTM5ck9Nb2hhbkJ1ZWJEZDBnMHFPLzdRa1JzYz0iLCJlbmRwb2ludHVybExlbmd0aCI6IjE2MSIsImlzbG9vcGJhY2siOiJUcnVlIiwiY2lkIjoiTW1JMU16RXpOR1V0TW1VMVl5MDBOVEJqTFRsa1lqQXRPREpoWVRrNVkyRmlZamRoIiwidmVyIjoiaGFzaGVkcHJvb2Z0b2tlbiIsInNpdGVpZCI6Ik1EVTROekk0TVdFdFpETTBOUzAwWWpjMExXRTJaamt0WkdFelpURXlNbUkzT1RSbSIsImFwcF9kaXNwbGF5bmFtZSI6InNhc2FzbCIsImFwcGlkIjoiOGEzYmFhYzktOGJmMy00MmJhLTg3OGEtYTNhMjk4ZTBlZDZiIiwidGlkIjoiYTU5Yjc0NzMtZWQyZi00ZDliLTgxYWQtN2U2MThjZTRjM2ZhIiwidXBuIjoib2xpaGV5QHJheWdpbmEub25taWNyb3NvZnQuY29tIiwicHVpZCI6IjEwMDNCRkZEQTJDMTM0RjAiLCJzY3AiOiJteWZpbGVzLnJlYWQgYWxsZmlsZXMucmVhZCBteWZpbGVzLndyaXRlIGFsbGZpbGVzLndyaXRlIiwidHQiOiIyIiwidXNlUGVyc2lzdGVudENvb2tpZSI6bnVsbH0.dHUyQXhuS2lXWFJQL2h3VEV3TitWT0NVNG9DMS9ZWkdTZTBZck82b1N3MD0 HTTP/1.1
> Host: raygina-my.sharepoint.com
> User-Agent: curl/7.47.0
> Accept: */*
>
< HTTP/1.1 401 Unauthorized
< Cache-Control: private
< Transfer-Encoding: chunked
< Content-Type: text/html; charset=utf-8
< P3P: CP="ALL IND DSP COR ADM CONo CUR CUSo IVAo IVDo PSA PSD TAI TELo OUR SAMo CNT COM INT NAV ONL PHY PRE PUR UNI"
< WWW-Authenticate: Bearer realm="a59b7473-ed2f-4d9b-81ad-7e618ce4c3fa",client_id="00000003-0000-0ff1-ce00-000000000000",trusted_issuers="00000001-0000-0000-c000-000000000000@*,https://sts.windows.net/*/,00000003-0000-0ff1-ce00-000000000000@90140122-8516-11e1-8eff-49304924019b",authorization_uri="https://login.windows.net/common/oauth2/authorize"
< x-ms-diagnostics: 3000001;reason="This access token is not valid on this endpoint.";category="invalid_request"
< SharePointError: 0
< X-AspNet-Version: 4.0.30319
< SPRequestGuid: b56d479e-7088-4000-c9cd-aed0981f3da2
< request-id: b56d479e-7088-4000-c9cd-aed0981f3da2
< MS-CV: nkdttYhwAEDJza7QmB89og.0
< Strict-Transport-Security: max-age=31536000
< X-Powered-By: ASP.NET
< MicrosoftSharePointTeamServices: 16.0.0.7324
< X-Content-Type-Options: nosniff
< X-MS-InvokeApp: 1; RequireReadOnly
< X-MSEdge-Ref: Ref A: 14DAABCA3CB64002BBE51B07950244F0 Ref B: AMSEDGE1220 Ref C: 2018-02-05T06:36:53Z
< Date: Mon, 05 Feb 2018 06:36:53 GMT
<
* Connection #0 to host raygina-my.sharepoint.com left intact
{"error":"invalid_request","error_description":"This access token is not valid on this endpoint."}
```

**Comment by ificator** [comment]
Unfortunately the reliance on the order of the parameters for these cases was a sacrifice to support two scenarios:
1. URL integrity (we do not want our query parameters modified)
2. The flexibility to add other query string parameters to the end of the URL if clients need to do so

If we allowed ordering to not matter, we would not able to differentiate between parameters that cannot be modified and parameters that were added after the fact and so would not be able to support the second scenario.

Having said all that, we realize that there are some frameworks that do have a habit of reordering query string parameters (normally sorting alphabetically) and so if that's causing issues for you we'd suggest making sure there's a suggestion over at https://onedrive.uservoice.com/forums/262982-onedrive?category_id=89523
